src/pages/iou/WaypointEditor.js

```diff
@@ -0,0 +1,161 @@
+import React, {useRef} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import AddressSearch from '../../components/AddressSearch';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../components/HeaderWithBackButton';
+import Navigation from '../../libs/Navigation/Navigation';
+import ONYXKEYS from '../../ONYXKEYS';
+import Form from '../../components/Form';
+import styles from '../../styles/styles';
+import compose from '../../libs/compose';
+import CONST from '../../CONST';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import * as Transaction from '../../libs/actions/Transaction';
+import * as ValidationUtils from '../../libs/ValidationUtils';
+import ROUTES from '../../ROUTES';
+import {withNetwork} from '../../components/OnyxProvider';
+import networkPropTypes from '../../components/networkPropTypes';
+
+const propTypes = {
+    /** The transactionID of the IOU */
+    transactionID: PropTypes.string.isRequired,
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** IOU type */
+            iouType: PropTypes.string,
+
+            /** Index of the waypoint being edited */
+            waypointIndex: PropTypes.string,
+        }),
+    }),
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    route: {
+        params: {
+            iouType: '',
+            waypointIndex: '',
+        },
+    },
+};
+
+function WaypointEditor({transactionID, route: {params: {iouType = '', waypointIndex = ''} = {}} = {}, network, translate}) {
+    const textInput = useRef(null);
+
+    const validate = (values) => {
+        const errors = {};
+        const waypointValue = values[`waypoint${waypointIndex}`] || '';
+        if (network.isOffline && waypointValue !== '' && !ValidationUtils.isValidAddress(waypointValue)) {
+            errors[`waypoint${waypointIndex}`] = 'bankAccount.error.address';
+        }
+
+        // Require the user to select a suggested address if they are online
```

Comment:
The code below is a bit confusing so maybe let's elaborate on the comment a bit.
```
// If the user is online and they are trying to save a value without using the autocomplete, show an error message instructing them to use a selected address instead. That enables us to save the address with coordinates when it is selected
```